nav2_behaviors/include/nav2_behaviors/plugins/drive_on_heading.hpp

```diff
@@ -133,7 +134,35 @@ class DriveOnHeading : public TimedBehavior<ActionT>
     cmd_vel->header.frame_id = this->robot_base_frame_;
     cmd_vel->twist.linear.y = 0.0;
     cmd_vel->twist.angular.z = 0.0;
-    cmd_vel->twist.linear.x = command_speed_;
+    if (acceleration_limit_ <= 0.0 && deceleration_limit_ <= 0.0) {
+      cmd_vel->twist.linear.x = command_speed_;
+    } else {
+      double current_speed = last_vel_ ? *last_vel_ : 0.0;
```

Comment:
I think the behaviors have a odom subscriber to use for the current velocity. 

This actually brings up a whole topic about whether we should base the velocity closed loop (on odom speed) or open looop (on the last command sent, assuming it was executed). We have this also topic in the controllers and velocity smoother. If we're adding in dynamic constraints here, I think its sensible we decide what makes sense here as well